packages/clerk-js/src/ui/components/SignIn/SignInFactorOnePasswordCard.tsx

```diff
@@ -8,25 +8,60 @@ import { Card, CardAlert, Footer, Form, Header, IdentityPreview, useCardState }
 import { useSupportEmail } from '../../hooks/useSupportEmail';
 import { useRouter } from '../../router/RouteContext';
 import { handleError, useFormControl } from '../../utils';
+import { HavingTrouble } from './HavingTrouble';
 
 type SignInFactorOnePasswordProps = {
-  onShowAlternativeMethodsClick: React.MouseEventHandler;
+  onShowAlternativeMethodsClick: React.MouseEventHandler | undefined;
   onFactorPrepare: (f: ResetPasswordCodeFactor) => void;
 };
 
-export const SignInFactorOnePasswordCard = (props: SignInFactorOnePasswordProps) => {
+const usePasswordControl = (props: SignInFactorOnePasswordProps) => {
   const { onShowAlternativeMethodsClick, onFactorPrepare } = props;
-  const card = useCardState();
-  const { setActive } = useCoreClerk();
   const signIn = useCoreSignIn();
-  const { navigateAfterSignIn } = useSignInContext();
-  const supportEmail = useSupportEmail();
+
   const passwordControl = useFormControl('password', '', {
     type: 'password',
     label: localizationKeys('formFieldLabel__password'),
   });
+
+  const resetPasswordFactor = signIn.supportedFirstFactors.find(
```

Comment:
Was reset password not available in this component before?

Reply:
It was, this seems like an addition, but I just grouped the pre-existed logic in a hook.